First implementation of Eclipse Marketplace dp version check

nicolatimeus opened the pull request:
See #1032

Signed-off-by: nicolatimeus <nicola.timeus@eurotech.com>

nicolatimeus committed:
First implementation of Eclipse Marketplace dp version check

Signed-off-by: nicolatimeus <nicola.timeus@eurotech.com>

MMaiero commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/Packages/PackagesPanelUi.java:
```diff
@@ -122,6 +127,24 @@
     @UiField
     Well marketplaceInstallWell;
 
+    @UiField
```
Can these UIField be private?
Maybe also marketplaceDescriptor can be private aswell

MMaiero commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/Packages/PackagesPanelUi.java:
```diff
@@ -448,48 +482,57 @@ public void onSuccess(GwtXSRFToken token) {
 
-    private void initModal() {
+    private void initModals() {
         this.uploadErrorModal.setTitle(MSGS.warning());
```
let's add two private methods for respective modals initialization.

MMaiero commented:
@nicolatimeus I've rebased the branch and made minor changes. Please have a look.

MMaiero committed:
Added compatibility property to kura.properties file.

The property is enabled by default and filled when building Kura.
Applied minor changes to the javadoc.
Minor changes to SystemServiceImpl.java, GwtPackageServiceImpl.java and GwtKuraErrorCode.java.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

nicolatimeus commented:
The changes seem fine

MMaiero commented:
Ok, Merging then.
